File: Python/underautomation/universal_robots/rtde/internal/rtde_parameters_base.py
import typing
from underautomation.universal_robots.rtde.rtde_versions import RtdeVersions
from underautomation.universal_robots.rtde.rtde_output_setup import RtdeOutputSetup
from underautomation.universal_robots.rtde.rtde_input_setup import RtdeInputSetup
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Rtde.Internal import RtdeParametersBase as rtde_parameters_base

class RtdeParametersBase:
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = rtde_parameters_base()
		else:
			self._instance = _internal
	@property
	def frequency(self) -> float:
		return self._instance.Frequency
	@frequency.setter
	def frequency(self, value: float):
		self._instance.Frequency = value
	@property
	def version(self) -> RtdeVersions:
		return RtdeVersions(self._instance.Version)
	@version.setter
	def version(self, value: RtdeVersions):
		self._instance.Version = value
	@property
	def output_setup(self) -> RtdeOutputSetup:
		return RtdeOutputSetup(self._instance.OutputSetup)
	@output_setup.setter
	def output_setup(self, value: RtdeOutputSetup):
		self._instance.OutputSetup = value
	@property
	def input_setup(self) -> RtdeInputSetup:
		return RtdeInputSetup(self._instance.InputSetup)
	@input_setup.setter
	def input_setup(self, value: RtdeInputSetup):
		self._instance.InputSetup = value

File: Rtde/Internal/RtdeParametersBase.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Rtde;

namespace Rtde.Internal {
	/// <summary>
	/// Base parameters to set up RTDE
	/// </summary>
	public abstract class RtdeParametersBase {

		/// <summary>
		/// For RTDE version 2, you can specify a frequency for output received data. Maximum frequency depends on your robot version.
		/// If you set frequency to 0, maximum frequency will be choosen
		/// Default value is 10Hz
		/// </summary>
		public double Frequency { get; set; }

		/// <summary>
		/// RTDE version. If set to Auto, the most recent version will be choosen according to your robot version
		/// Default value is V2
		/// </summary>
		public RtdeVersions Version { get; set; }

		/// <summary>
		/// List of all output data the robot will send to your application
		/// </summary>
		public RtdeOutputSetup OutputSetup { get; set; }

		/// <summary>
		/// List of all input data you can send to the robot
		/// </summary>
		public RtdeInputSetup InputSetup { get; set; }

		/// <summary>
		/// Default RTDE TCP port used (30004)
		/// </summary>
		public const int DEFAULT_PORT = 30004;

		/// <summary>
		/// TCP port used for RTDE connection. Default : 30004
		/// </summary>
		public int Port { get; set; }
	}
}
